sympy/solvers/ode.py

```diff
@@ -363,24 +359,13 @@ def sub_func_doit(eq, func, new):
     x*(-1/(x**2*(z + 1/x)) + 1/(x**3*(z + 1/x)**2)) + 1/(x*(z + 1/x))
     ...- 1/(x**2*(z + 1/x)**2)
     """
-    reps = {}
-    repu = {}
+    reps= {func: new}
```

Comment:
Ahhh....this line is not necessary. This is forcing unecessary specificity into the equation to be tested against zero. If f(x) is going to cancel, it doesn't matter what form it has, so it can be left as f(x). The derivatives, however, need to be calculated. So when you make your next mods, could you just make that `reps = {}` and see if it works locally before pushing it here?